### PR TITLE
AKCORE-3: ShareGroupHeartbeat RPC implementation using group coordinator functionality

### PR DESCRIPTION
**Pull request by adixitconfluent**
**About**
This PR completed the 2 TODOs mentioned in this [PR](https://github.com/confluentinc/kafka/pull/1003)- 
1. Send the ShareGroupHeartbeatRequest to the group coordinator and receive the response as ShareGroupHeartbeatResponse
2. Uncomment the test which verifies the functionality mentioned in point 1

Reference -[ AKCORE-3](https://confluentinc.atlassian.net/browse/AKCORE-3)

**Build** 
The build is passing as shown in the screenshot
Screenshot 2024-02-07 at 5 06 12 PM

The check styles is passing as shown in the screenshot
Screenshot 2024-02-07 at 5 06 36 PM

**Testing**
The uncommented test passes as shown in the sacreenshot
Screenshot 2024-02-07 at 5 06 59 PM

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaApis.scala [hunk 1]
```diff
@@ -3816,8 +3816,6 @@ class KafkaApis(val requestChannel: RequestChannel,
       requestHelper.sendMaybeThrottle(request, shareGroupHeartbeatRequest.getErrorResponse(Errors.GROUP_AUTHORIZATION_FAILED.exception))
       CompletableFuture.completedFuture[Unit](())
     } else {
-      /* TODO: Once ShareGroupHeartBeatAPI is supported in group coordinator, uncomment the code below
-          (lines 3821 - 3833) and remove the part sending INVALID_CONFIG exception (lines 3835, 3836)
```
Comment (apoorvmittal10) [inline core/src/main/scala/kafka/server/KafkaApis.scala:3820]: Do we need this log line `info(s"*** SHARE GROUP HEARTBEAT RESPONSE *** + ${response}")` on every reponse in info mode? Maybe debug?

**Review by apoorvmittal10 — APPROVED** [review]
LGTM. Can we have concise PR title, seems like a description to me.

File: core/src/main/scala/kafka/server/KafkaApis.scala (hunk 1 above)
Comment (adixitconfluent) [reply core/src/main/scala/kafka/server/KafkaApis.scala:3820]: my bad, I was using it during my testing. Will remove it now.

**Comment by adixitconfluent** [comment]
> LGTM. Can we have concise PR title, seems like a description to me.

yes, I have changed the description appropriately

**Pull request merged**